scripts/build_codelabs

```diff
@@ -32,24 +34,126 @@ function is_xcpretty_available() {
   fi
 }
 
+# Check that flags have arguments.
+#
+# Exits if the flag does not have an argument.
+function check_arguments() {
+  if [ -z $1 ]; then
+    echo "Argument must be provided to flag."
+    exit 1
+  fi
+}
+
+# Checks that the argument is a valid reference.
+#
+# Exits if the reference is not valid.
+function check_reference() {
+  reference=$(git ls-remote https://github.com/material-components/material-components-ios.git $1 | wc -l)
+  if [ $reference -eq 0 ]; then
+    echo "Invalid reference name."
+    exit 1
+  fi
+}
+
+# Checks that the argument is a valid commit.
+#
+# Exits if the commit is not valid.
+function check_commit() {
+  if [ ! $(git cat-file commit $1) ]; then
+    echo "Invalid commit SHA."
+    exit 1
+  fi
+}
+
 # Parse command-line arguments.
 #
 # Note that we're following the command-line exit status convention of zero
 # to mean "success."
 verbose=1
-for i in "$@"; do
-  case $i in
+all_codelabs=0
+has_source=0
+
+# Set default branch to this branch
+export DEFAULT_BRANCH=$(git symbolic-ref --short HEAD)
+
+while [ $# -gt 0 ]; do
+  case $1 in
     -v|--verbose)
       verbose=0
       shift
       ;;
+    -rb|--remote-branch)
+      shift
+      check_arguments $1
+      check_reference $1
+      if [ $has_source -eq 1 ]; then
+        echo "Only one of --remote-branch/--commit/--tag allowed."
+        exit -1
+      fi
+      export TEST_BRANCH=$1
+      has_source=1
+      shift
+      ;;
+    -c|--commit)
+      shift
+      check_arguments $1
+      check_commit $1
+      if [ $has_source -eq 1 ];then
+        echo "Only one of --remote-branch/--commit/--tag allowed."
+        exit -1
+      fi
+      export TEST_COMMIT=$1
+      has_source=1
+      shift
+      ;;
+    -t|--tag)
+      shift
+      check_arguments $1
+      check_reference $1
+      if [ $has_source -eq 1 ];then
+        echo "Only one of --remote-branch/--commit/--tag allowed."
+        exit -1
+      fi
+      export TEST_TAG=$1
+      has_source=1
+      shift
+      ;;
+    -a|--all)
+      all_codelabs=1
+      shift
+      ;;
+    -h|--help)
+      echo
+      echo "Builds MDC iOS Codelabs."
+      echo
+      echo "The default branch of MDC iOS that is used for testing is your current branch."
+      echo "Your current branch is: $DEFAULT_BRANCH"
+      echo
+      echo "The default codelabs built are 104 Complete and 111 Complete. You can build all with -a."
```

Comment:
Consider:
By default the Swift and Objective-C Google I/O 2018 Codelab 104 and Codelab 111 targets will be built.
You can build all codelabs by specifying --all.

Reply:
Right now, the script points to the master branch of the codelabs (I can't see an I/O branch there). Should I refer to the codelabs as the Google I/O 2018 codelabs?

Reply:
Good point. 
Could you add tag to the current master called googleio2018 at head? (Outside of this PR)
That way we can continue to point at master, but we'll have a reference if we need it.
I think referring to them as the 'codelabs' is fine in the general case, in this specific case I would urge details.  My thinking is that if something goes wrong, the person will end up coming through this path (--help !) and will gain some insight.